tket/src/Mapping/RoutingMethodCircuit.cpp

```diff
@@ -54,30 +37,33 @@ unit_map_t RoutingMethodCircuit::routing_method(
       mapping_frontier->get_default_to_quantum_boundary_unit_map());
 
   // get routed subcircuit
-  std::tuple<Circuit, unit_map_t, unit_map_t> routed_subcircuit =
+  std::tuple<bool, Circuit, unit_map_t, unit_map_t> routed_subcircuit =
       this->route_subcircuit_(frontier_circuit, architecture);
-  unit_map_t new_labelling = std::get<1>(routed_subcircuit);
+
+  if (!get<0>(routed_subcircuit)) {
+    return {false, {}};
+  }
 
   // update unit id at boundary in case of relabelling
-  mapping_frontier->update_quantum_boundary_uids(new_labelling);
+  mapping_frontier->update_quantum_boundary_uids(get<2>(routed_subcircuit));
 
   unit_map_t swap_permutation;
-  for (const auto& pair : new_labelling) {
+  for (const auto& pair : get<2>(routed_subcircuit)) {
     if (pair.first != pair.second &&
         architecture->node_exists(Node(pair.first))) {
       swap_permutation.insert(pair);
     }
   }
   // permute edges held by unitid at out boundary due to swaps
   mapping_frontier->permute_subcircuit_q_out_hole(
-      std::get<2>(routed_subcircuit), frontier_subcircuit);
+      get<3>(routed_subcircuit), frontier_subcircuit);
```

Comment:
I'm not sure how this builds: I couldn't find a `using` declaration for `std::get` anywhere in our code, so I guess it is coming from a header in some dependency. Anyway, can we please use `std::get` rather than rely on an accidental inclusion? Or put `using std::get;` at the top of this file.

Reply:
Changed